Appdata related patches

sabriunal opened the pull request:
### appdata: Use appstreamcli for appdata validation

appstream-util is obsoleted by appstreamcli.

### data: Mark two strings as untranslatable

Mark the developer name and the application name as untranslatable.

### appdata: Update appdata

- Remove one of Purism tag to pass validation
- Add the launchable tag
- Add translate URL
- Add keywords and categories that in use GNOME Software and KDE Discover

### appdata: Use longer sentence for app description](appdata: Use longer sentence for app description

Fix appstreamcli warning
    
"Foliate is modern e-book reader app designed for GNOME.
The first `description/p` paragraph of this component might be too short (< 80 characters). Please consider starting with a longer     paragraph to improve how the description looks like in software centers and to provide more detailed information on this component immediately"

yakushabb committed:
appdata: Use appstreamcli for appdata validation

appstream-util is obsoleted by appstreamcli.

yakushabb committed:
data: Mark two strings as untranslatable

Mark the developer name and the application name as untranslatable.

yakushabb committed:
appdata: Update appdata

- Remove one of Purism tag to pass validation
- Add the launchable tag
- Add translate URL
- Add keywords and categories that in use GNOME Software and KDE Discover

johnfactotum commented:
> Mark the developer name and the application name as untranslatable.

IMHO, whether names are translated depends on the locale and should be left to the discretion of the translator. 

Generally speaking, it's not uncommon to transcribe or translate names, especially when the target language is in a different writing system.

> Add keywords and categories that in use GNOME Software and KDE Discover

Do they not pull those from the desktop file if there's a `<launchable>` ?

> Use longer sentence for app description

The current description is indeed not that great. But I must say I'm not really happy with the new version, either.

sabriunal commented on data/com.github.johnfactotum.Foliate.metainfo.xml.in:
```diff
@@ -3,11 +3,11 @@
     <translation type="gettext">com.github.johnfactotum.Foliate</translation>
     <description>
-        <p>Foliate is modern e-book reader app designed for GNOME.</p>
+        <p>Dive into a literary journey with Foliate, your friendly GNOME companion for e-books! From EPUB to PDF, enjoy personalized reading with themes, bookmarks, and quick lookup.</p>
```
Seems a bit awkward.

sabriunal replied:
Some alternatives as suggestion:

> Foliate is a modern e-book reader app designed for GNOME and is capable of opening EPUB, Mobipocket, Kindle, FB2, CBZ, and PDF files.

> Foliate is a modern e-book reader app designed for GNOME. From EPUB to PDF, enjoy personalized reading with themes, bookmarks, and quick lookup.

> Foliate is a modern e-book reader app designed for GNOME. From EPUB to PDF, enjoy personalized reading with themes, bookmarks, and quick lookup.

sabriunal commented:
> > Add keywords and categories that in use GNOME Software and KDE Discover
> 
> Do they not pull those from the desktop file if there's a `<launchable>` ?

Seems a bit backwards compatibility but yes both of them support.
- GNOME Software [comment](https://gitlab.gnome.org/GNOME/gnome-software/-/issues/2375#note_1909776) 
- KDE Discover [comment](https://discuss.kde.org/t/does-kde-discover-parse-the-desktop-file-for-categories-and-keywords/7041/2)

But the 'keywords' tag is [official](https://freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-keywords)

> > Mark the developer name and the application name as untranslatable.
> 
> IMHO, whether names are translated depends on the locale and should be left to the discretion of the translator.
> 
> Generally speaking, it's not uncommon to transcribe or translate names, especially when the target language is in a different writing system.

I think your name is written in Latin. Yes, translatable names are useful for Arabic, Chinese, Japanese, etc., names.

johnfactotum commented:
Hmm. I guess the engaging tone of "Dive into a literary journey with Foliate..." does sound better. But maybe de-emphasize specific features as those are listed below. How about

> Discover a new chapter in reading with Foliate, the modern e-book reader tailored for GNOME. Immerse yourself in a distraction-free interface, with customization features designed to match your unique preferences.

yakushabb committed:
appdata: Use longer sentence for app description

Fix appstreamcli warning

"Foliate is modern e-book reader app designed for GNOME.

The first `description/p` paragraph of this component might be
too short (< 80 characters). Please consider starting with a longer
paragraph to improve how the description looks like in software centers
and to provide more detailed information on this component immediately
in the first paragraph."

sabriunal commented:
> Hmm. I guess the engaging tone of "Dive into a literary journey with Foliate..." does sound better. But maybe de-emphasize specific features as those are listed below. How about
> 
> > Discover a new chapter in reading with Foliate, the modern e-book reader tailored for GNOME. Immerse yourself in a distraction-free interface, with customization features designed to match your unique preferences.

Updated.

johnfactotum commented:
Thanks!
